doc/guides/logging/dictionary.rst

```diff
@@ -0,0 +1,208 @@
+.. _logging_guide_dictionary:
+
+Dictionary-based Logging
+########################
+
+Dictionary-based logging, instead of human readable texts, outputs the log
+messages in binary format. This binary format encodes arguments to formatted
+strings in their native storage formats which can be more compact than their
+text equivalents. For statically defined strings (including the format
+strings and any string arguments), references to the ELF file are encoded
+instead of the whole strings. A dictionary created at build time contains
+the mappings between these references and the actual strings. This allows
+the offline parser to obtain the strings from the dictionary to parse
+the log messages. This binary format allows a more compact representation
+of log messages in certain scenarios. However, this requires the use of
+an offline parser and is not as intuitive to use as text-based log messages.
+
+Note that ``long double`` is not supported by Python's ``struct`` module.
+Therefore, log messages with ``long double`` will not display the correct
+values.
+
+
+Configuration
+*************
+
+Here are kconfig options related to dictionary-based logging:
+
+- :option:`CONFIG_LOG_DICTIONARY_SUPPORT` enables dictionary-based logging
+  support. This should be selected by the backends which require it.
+
+- :option:`CONFIG_LOG_DICTIONARY_BACKEND_UART` enables the UART backend
+  for dictionary-based logging.
+
+  - :option:`CONFIG_LOG_DICTIONARY_BACKEND_UART_HEX` tells the UART backend
+    to output hexidecimal characters instead of binary data. This is useful
+    when the log data needs to be captured manually via terminals and
+    consoles.
+
+  - :option:`CONFIG_LOG_DICTIONARY_BACKEND_UART_BIN` tells the UART backend
+    to output binary data.
+
+
+Usage
+*****
+
+When dictionary-based logging is enabled via enabling related logging backends,
+a JSON database file, named :file:`log_dictionary.json`, will be created
+in the build directory. This database file contains information for the parser
+to correctly parse the log data. Note that this database file only works
+with the same build, and cannot be used for any other builds.
+
+To use the log parser:
+
+.. code-block:: console
+
+   ./scripts/logging/dictionary/log_parser.py <build dir>/log_dictionary.json <log data file>
+
+The parser takes two required arguments, where the first one is the full path
+to the JSON database file, and the second part is the file containing log data.
+Add an optional argument ``--hex`` to the end if the log data file contains
+hexidecimal characters
+(e.g. when ``CONFIG_LOG_DICTIONARY_BACKEND_UART_HEX=y``). This tells the parser
+to convert the hexidecimal characters to binary before parsing.
+
+
+Example
+-------
+
+Here is an example of how to use dictionary-based logging.
+
+1. :file:`samples/hello_world/src/main.c` can be modified for demostration
+   purpose.
```

Comment:
Instead of instructions to modify an existing sample, why not create a new sample in `samples/subsys/logging` and have all the necessary code there.

Reply:
Having a new sample app is better... what was I thinking...? :)